docs/website/root/mithril/threat-model.md

```diff
@@ -0,0 +1,318 @@
+
+:::
+
+We then identify threats and countermeasures
+
+#### KES private keys
+
+* KES key is present only on BP Node but needs to be shared with both the cardano-node process and the mithril-signer process
+* KES keys are needed by mithril-signer in order to sign verification key along with an operational certificate which authenticates the key from this stake pool id
+* This signing happens at every epoch
+    * **availability**: Yes (If KES key is unavailable then signing cannot proceeed)
+    * **confidentiality**: Yes (Capturing KES private keys allow an attacker to impersonate a registered SPO on-chain and produce blocks on his behalf)
+    * **integrity**: Yes (Rotating a compromised KES key is a time-consuming process, not sure if it's even possible if the KES period has not passed completely? See https://github.com/input-output-hk/cardano-node-wiki/blob/main/docs/stake-pool-operations/7_KES_period.md)
+
+#### Block diffusion
+
+Block diffusion process ensures the timely diffusion of blocks, both those produced "locally" and those received from upstream peers
+* **availability**: Yes (Not being to diffuse blocks harms a BP's SPO economic viability, and can also harm their ability to create new blocks)
+* **confidentiality**: No (block diffusion happens in the open)
+* **integrity**: Yes (partially? not sure what integrity means here)
+
+#### Block production
+
+Block production is the process of "minting" new blocks by block producers, driven by Stake-based random lottery.
+
+The mithril-signer necessarily runs on the same host than a BP because it needs access to the KES signing key
+
+* **availability**: Yes (BP is critical for SPOs revenue, and preventing a BP from producing blocks can harm SPOs capabilities to operate)
+* **integrity**: Yes (incorrect or invalid data can hamper BP capabilities)
+* **confidentiality**: Yes? (BP schedule is private information for the BP, leaking it could provide adversaries advance knowledge of schedules and lead to **grinding attacks** to attempt to manipulate nonce in the disfavour of a BP ??)
+
+#### Cardano Chain Database
+
+A cardano-node maintains an on-disk database consisting of the chain's history. This database is updated by the node when new blocks are diffused through the network, or minted, and also contains a cache of the ledger state. 
+
+Mithril signer needs needs access to _trusted_ and _up-to-date_ Chain database in order to be able to sign snapshots. 
+
+* **availability**: Yes
+* **integrity**: Yes
+* **Confidentiality**: No (Data is public and replicated) 
+
+#### Cardano Ledger state
+
+Access to an accurate ledger state is needed by Mithril signer to retrieve reliable _Stake distribution_. This access is currently done through a local connection (direct w/ Pallas or indirect with cardano-cli) to a trusted cardano-node
```

Comment:
The ledger state / stake distribution is also used by the cardano-node to determine leader schedules and hence corruption here has an impact on the block production process.

Reply:
Access by the signer is done through local state query mini-protocols of the Cardano node (via local connection or Cardano cli). IMO, there is no impact on the leader schedule or block production process in that case. WDYT?

Reply:
I added your sentence to the section